includes/class-wc-product-variation.php

```diff
@@ -283,6 +283,23 @@ public function get_image_id( $context = 'view' ) {
 		return $image_id;
 	}
 
+	/**
+	 * Get shipping class ID.
+	 *
+	 * @since 2.7.0
+	 * @param  string $context
+	 * @return int
+	 */
+	public function get_shipping_class_id( $context = 'view' ) {
+		$shipping_class_id = $this->get_prop( 'shipping_class_id', $context );
+
+		if ( 'view' === $context && ! $shipping_class_id ) {
+			$shipping_class_id = $this->parent_data['shipping_class_id'];
```

Comment:
Should `shipping_class_id`, `title`, `backorders`, and `image_id` keys be added to the protected $parent_data field (line 29)? The data store will fill these keys in with `read_product_data`, but the WC_Product_Variation class on its own will throw undefined index notices since they're not set.